python/paddle/fluid/tests/unittests/test_multimarginloss.py

```diff
@@ -0,0 +1,348 @@
+        if weight_np is not None:
+            weight = paddle.static.data(name='weight',
+                                        shape=weight_np.shape,
+                                        dtype=weight_np.dtype)
+            feed_dict['weight'] = weight_np
+        if functional:
+            res = call_MultiMarginLoss_functional(input=input,
+                                                  label=label,
+                                                  p=p,
+                                                  margin=margin,
+                                                  weight=weight,
+                                                  reduction=reduction)
+        else:
+            res = call_MultiMarginLoss_layer(input=input,
+                                             label=label,
+                                             p=p,
+                                             margin=margin,
+                                             weight=weight,
+                                             reduction=reduction)
+
+        exe = paddle.static.Executor(place)
+        static_result = exe.run(prog, feed=feed_dict, fetch_list=[res])
+    return static_result
+
+
+def test_dygraph(place,
+                 input,
+                 label,
+                 p=1,
+                 margin=1.0,
+                 weight=None,
+                 reduction='mean',
+                 functional=False):
+    paddle.disable_static()
+    input = paddle.to_tensor(input)
+    label = paddle.to_tensor(label)
+
+    if weight is not None:
+        weight = paddle.to_tensor(weight)
+    if functional:
+        dy_res = call_MultiMarginLoss_functional(input=input,
+                                                 label=label,
+                                                 p=p,
+                                                 margin=margin,
+                                                 weight=weight,
+                                                 reduction=reduction)
+    else:
+        dy_res = call_MultiMarginLoss_layer(input=input,
+                                            label=label,
+                                            p=p,
+                                            margin=margin,
+                                            weight=weight,
+                                            reduction=reduction)
+    dy_result = dy_res.numpy()
+    paddle.enable_static()
+    return dy_result
+
+
+def calc_multi_margin_loss(
+    input,
+    label,
+    p=1,
+    margin=1.0,
+    weight=None,
+    reduction='mean',
+):
+    label = label.reshape(-1, 1)
+    index_sample = []
+    for i in range(len(label)):
+        index_sample.append(input[i, label[i]])
+    index_sample = np.array(index_sample).reshape(-1, 1)
+
+    if weight is None:
+        expected = np.mean(np.maximum(margin + input - index_sample, 0.0)**p,
+                           axis=1) - margin**p / input.shape[1]
+    else:
+        weight = weight.reshape(-1, 1)
```

Comment:
weight should be for each class , not for each example.

Reply:
Done